src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Performance/PreferHashDataOverComputeHashTests.cs

```diff
@@ -0,0 +1,1188 @@
+        public async Task BasicObjectCreationUsingBlockCase()
+        {
+            string vbInput = @"
+Imports System
+Imports System.Security.Cryptography
+
+Public Class Test
+    Public Shared Sub TestMethod()
+        Dim buffer = New Byte(1023) {}
+        {|#2:Using sha256 As New SHA256Managed()|}
+            Dim aboveLine = 20
+            Dim digest As Byte() = {|#0:sha256.ComputeHash({|#1:buffer|})|}
+            Dim belowLine = 10
+        End Using
+    End Sub
+End Class
+";
+
+            string vbFix = @"
+Imports System
+Imports System.Security.Cryptography
+
+Public Class Test
+    Public Shared Sub TestMethod()
+        Dim buffer = New Byte(1023) {}
+        Dim aboveLine = 20
+        Dim digest As Byte() = SHA256.HashData(buffer)
+        Dim belowLine = 10
+    End Sub
+End Class
+";
+            await TestVBWithVariable(
+                vbInput,
+                vbFix,
+                "System.Security.Cryptography.SHA256");
+        }
+
+        [Fact]
+        public async Task CSharpObjectCreationUsingStatementCastedCase()
+        {
+            string csInput = @"
+using System;
+using System.Security.Cryptography;
+
+public class Test
+{
+    public static void TestMethod()
+    {
+        var buffer = new byte[1024];
+        using ({|#2:HashAlgorithm sha256 = new SHA256Managed()|})
+        {
+            int aboveLine = 20;
+            byte[] digest = {|#0:sha256.ComputeHash({|#1:buffer|})|};
+            int belowLine = 10;
+        }
+    }
+}
+";
+
+            string csFix = @"
+using System;
+using System.Security.Cryptography;
+
+public class Test
+{
+    public static void TestMethod()
+    {
+        var buffer = new byte[1024];
+        int aboveLine = 20;
+        byte[] digest = SHA256.HashData(buffer);
+        int belowLine = 10;
+    }
+}
+";
+            await TestCSWithVariable(
+                csInput,
+                csFix,
+                "System.Security.Cryptography.SHA256");
+        }
+
+        [Fact]
+        public async Task BasicObjectCreationUsingBlockCastedCase()
+        {
+            string vbInput = @"
+Imports System
+Imports System.Security.Cryptography
+
+Public Class Test
+    Public Shared Sub TestMethod()
+        Dim buffer = New Byte(1023) {}
+        {|#2:Using sha256 As HashAlgorithm = New SHA256Managed()|}
+            Dim aboveLine = 20
+            Dim digest As Byte() = {|#0:sha256.ComputeHash({|#1:buffer|})|}
+            Dim belowLine = 10
+        End Using
+    End Sub
+End Class
+";
+
+            string vbFix = @"
+Imports System
+Imports System.Security.Cryptography
+
+Public Class Test
+    Public Shared Sub TestMethod()
+        Dim buffer = New Byte(1023) {}
+        Dim aboveLine = 20
+        Dim digest As Byte() = SHA256.HashData(buffer)
+        Dim belowLine = 10
+    End Sub
+End Class
+";
+            await TestVBWithVariable(
+                vbInput,
+                vbFix,
+                "System.Security.Cryptography.SHA256");
```

Comment:
All of these tests used SHA256.  The analyzer/fixer should also detect MD5, SHA1, SHA384, and SHA512.

Reply:
The tests use SHA256 only for convenience. Let me see how I can expand that in a nice pattern

Reply:
Updated

Reply:
The analyzer does specifically uses `SHA256.HashData` method as an early termination test but it is not tied to the analyzer search logic